src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1580,45 +1633,45 @@ private bool HasEntryPointSignature(MethodSymbol method, DiagnosticBag bag)
                 returnsTaskOrTaskOfInt = ReturnsAwaitableToVoidOrInt(method, bag);
                 if (!returnsTaskOrTaskOfInt)
                 {
-                    return false;
+                    return (false, false);
                 }
             }
 
             // Prior to 7.1, async methods were considered to "have the entrypoint signature".
             // In order to keep back-compat, we need to let these through if compiling using a previous language version.
             if (!returnsTaskOrTaskOfInt && method.IsAsync && this.LanguageVersion >= LanguageVersion.CSharp7_1)
             {
-                return false;
+                return (false, returnsTaskOrTaskOfInt);
```

Comment:
Are we ever using `IsTaskLike` if `IsCandidate` is `false`? Consider returning a single value rather than two separate `bool` values. Perhaps `enum EntryPointKind { NotCandidate, VoidOrInt, TaskOrTaskOfInt }`.